examples/wandb_sweeps/get_hyperparameters.py

```diff
@@ -0,0 +1,61 @@
+#!/usr/bin/env python
+"""Retrieves hyperparameters from a W&B run."""
+
+import argparse
+import logging
+
+import wandb
+import yaml
+
+import util
+
+# Top-level categories allowed in Lightning CLI configs.
+ALLOWED_TOP_LEVEL_CATEGORIES = frozenset(
+    ["data", "model", "trainer", "checkpoint", "prediction"]
+)
+
+
+def dot_to_nested_dict(flat_dict: dict[str, ...]) -> dict[str, ...]:
+    """Converts dot notation dictionary to nested dictionary.
+
+    Args:
+        flat_dict: Dictionary with dot-separated keys.
+
+    Returns:
+        Nested dictionary with proper structure.
+    """
+    nested = {}
+    for key, value in flat_dict.items():
+        if (
+            any(
+                key.startswith(f"{cat}.")
+                for cat in ALLOWED_TOP_LEVEL_CATEGORIES
+            )
+            or key in ALLOWED_TOP_LEVEL_CATEGORIES
+        ):
+            util.recursive_insert(nested, key, value)
+    return nested
+
+
+def main(args: argparse.Namespace) -> None:
+    api = wandb.Api()
+    run = api.run(f"{args.run_path}")
+    logging.info("Run URL: %s", run.url)
+    # Gets all config values from this run.
+    flat_config = {
+        key: value
+        for key, value in sorted(run.config.items())
```

Comment:
why are you sorting before making a dict? Does order matter here? I feel like if so, it would be better to use an explicitly order-preserving type

(I am aware dicts keep insert order in most modern python versions)